File: app/src/main/java/io/github/domi04151309/alwayson/DialogNls.java
package io.github.domi04151309.alwayson;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class DialogNls extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_nls);

        Button close_button = findViewById(R.id.okay_btn);
        close_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS"));
                finish();
            }
        });
    }
}

File: app/src/main/java/io/github/domi04151309/alwayson/DialogNls.kt
package io.github.domi04151309.alwayson

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class DialogNls : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_nls)

        val closeBtn = findViewById<Button>(R.id.okay_btn)
        closeBtn.setOnClickListener {
            startActivity(Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS"))
            finish()
        }
    }
}
